docs/io/ascii/howtoexamples.rst

```diff
@@ -0,0 +1,209 @@
+   >>> ascii.read(table, names=["Star", "spectral type"], data_start=1)
+   <Table length=2>
+    Star  spectral type
+    str6       str2
+   ------ -------------
+     Vega            A0
+   Altair            A7
+
+..
+  EXAMPLE END
+
+Badly formatted data line
+-------------------------
+
+Similar principles apply to badly formatted data lines. Here is a
+table where the number of columns is not consistent (`alpha Cen`
+should be written as `"alpha Cen"` to make clear that the two words
+"alpha" and "Cen" are part of the same column)::
+
+  Star SpT
+  Vega A0
+  alpha Cen G2V+K1
+
+When we try to read that, astropy throws an
+`astropy.io.ascii.InconsistentTableError` and prints out a long list of formats
+that it tried on this table and some advice (some output omitted)::
+
+  >>> from astropy.io import ascii
+  >>> table = '''
+  ... Star SpT
+  ... Vega A0
+  ... alpha Cen G2V+K1
+  ... '''
+  >>> ascii.read(table)
+  InconsistentTableError: Number of header columns (2) inconsistent with data columns (3) at data line 1
+  Header values: ['Star', 'SpT']
+  Data values: ['alpha', 'Cen', 'G2V+K1']
+  [...]
+
+This points us to the line with the problem, here line 1 (not counting the
+header lines and starting to count at 0 as usual in Python). In this table with
+just two lines that is easy to see, but for longer tables, the line number is
+very helpful. We can now fix that line by hand in the file or just comment it
+out (for most formats: putting `#` as the first character in the line) if we
+do not really need that line. Then, we can try to read the table again and see
+if it works or if there is a another badly formatted data line.
+
+Change a written table
+======================
+
+..
+  EXAMPLE START
+  Write table with end of line string
+
+If a single table needs to be written in a specific format that
+astropy does not support, one of the easiest solutions is to write a
+data file that is close to the desired format and then edit it by hand
+in a text editor. Often, we can add extra columns or place useful
+special values and then "search and replace all" in a text
+editor. Here, we want to write a table that says "end of line" at the
+end of every line. We can try to do that by simply adding another
+column to our table that has the content we are looking for::
+
+  >>> from astropy.table import Table
+  >>> from astropy.io import ascii
+  >>> table = Table({'a':[1, 2], 'b': ["some string", 4]})
+  >>> ascii.write(table)
+  a b
+  1 3
+  2 4
+  >>> table['end of line'] = 'end of line'
+  >>> ascii.write(table)
+  a b "end of line"
+  1 3 "end of line"
+  2 4 "end of line"
+
+We could now open this file in a text editor search and replace `"`
+with ``.  Instead, our new column could have simply been a values that
+does not occur in the table otherwise (e.g. "-99999") and then we
+could search and replace "-9999" with "end of line".
+
+..
+  EXAMPLE END
+
+
+Write our own reader or writer class
+====================================
+
```

Comment:
This is not a realistic option for the vast majority of users, having seen some recent work to add a new reader class, it pretty much requires help from the astropy devs. I think the existing https://docs.astropy.org/en/latest/io/ascii/read.html#advanced-customization section gets the point across.